Assets/Scripts/PowerUps/PowerUps/Shield.cs

```diff
@@ -11,12 +11,14 @@ public override void Activate()
         {
             base.Activate();
             GameService.Instance.GetPlayerService().GetPlayerController().SetShieldState(ShieldState.Activated);
+            GameService.Instance.GetPlayerService().GetPlayerController().ToggleShieldUI(true);
         }
```

Comment:
The method ToggleShieldUI indicates it will turn shield on if its off already and vice-versa.
SetShieldUI/SetShieldState is a more apropriate name.
using enums over booleans everytime. even if its a flag like shield on/ shield off, we will create an enum for it and use that instead of a boolean.
Enums are much more cleaner and readable.
The code of these projects you are making is going to be read thousand of people, so always have readability on back of your mind while writing code.